pgx-tests/src/framework.rs

```diff
@@ -56,14 +55,61 @@ where
     SHUTDOWN_HOOKS.lock().unwrap().push(Box::new(func));
 }
 
+// The goal of this closure is to allow "wrapping" of anything that might issue
+// an SQL simple_quuery or query using either a postgres::Client or
+// postgres::Transction and capture the output. The use of this wrapper is
+// completely optional, but it might help narrow down some errors later on.
+fn query_wrapper<F, T>(
+    query: Option<String>,
+    query_params: Option<&[&(dyn postgres::types::ToSql + Sync)]>,
+    mut f: F,
+) -> eyre::Result<T>
+where
+    T: IntoIterator,
+    F: FnMut(
+        Option<String>,
+        Option<&[&(dyn postgres::types::ToSql + Sync)]>,
+    ) -> Result<T, postgres::Error>,
+{
+    let result = f(query.clone(), query_params.clone());
+
+    match result {
+        Ok(result) => Ok(result),
+        Err(e) => {
+            let e = e.as_db_error().unwrap();
+            Err(eyre!(
+                "Postgres query failed:
+Query: {},
+Query Params: {:?},
+Error Message: {}
+Code: {:?}
+Severity: {}
+Detail: {}
+Hint: {}
+Schema: {}
+Table: {}",
```

Comment:
I feel like there might be a better way to do this?

Reply:
Well, it's somewhat common to use lowercase for framing text like this in Rust error messages so as not to distract from the more important (and possibly semantically capitalized) error messaging text. And we're still definitely adding a lot of vertical height to query failure, and a lot more for things that may be "None".

Query, yes. Error message, yes.

For the others... can it be made more compact so it doesn't add newlines when those are used? We're on an error path so we can afford to spend extra compute on formatting.

I also think that instead of `{}` for each case you should probably be assigning keys to these, so sev = e.severity(), and then using it up there somewhere.